partiql-eval/src/main/kotlin/org/partiql/eval/internal/operator/rex/ExprStructPermissive.kt

```diff
@@ -0,0 +1,38 @@
+package org.partiql.eval.internal.operator.rex
+
+import org.partiql.eval.internal.Environment
+import org.partiql.eval.internal.operator.Operator
+import org.partiql.eval.value.Datum
+import org.partiql.types.PType
+
+internal class ExprStructPermissive(private val fields: List<ExprStructField>) : Operator.Expr {
+    override fun eval(env: Environment): Datum {
+        val fields = fields.mapNotNull {
+            val key = it.key.eval(env)
+            if (key.isNull) {
+                return Datum.nullValue()
```

Comment:
Why is `null` returned for the overall struct constructor expression if one key is `null`? For the following query,

```
SELECT VALUE {v.a: v.b} FROM [{'a':'legit', 'b':1}, {'a':null, 'b':2}] AS v
```

`<< { legit: 1 }, null >>` will currently be the output.

My read of the "Treatment of mistyped attribute names" section of the PartiQL spec is that `null` struct key values (alike non-strings and `missing` values) would just omit that struct field from the struct in permissive mode. Following my read, the above query would result in `<< { 'legit': 1 }, {} >>`.

Reply:
Ok, nice! From our offline discussion, I've created an issue with the team's consensus on behavior as well as the need for potential clarification in the PartiQL Spec: https://github.com/partiql/partiql-lang/issues/88

I've just updated this PR accordingly. I also needed to rebase off of the type-renaming PR. The last two commits represent what has been updated.